recipes/full_finetune_distributed.py

```diff
@@ -550,18 +589,24 @@ def save_checkpoint(
         )
 
         if intermediate_checkpoint:
-            opt_state_dict = training.get_full_optimizer_state_dict(
-                self._optimizer,
-                self._is_rank_zero,
-                device=self._device,
-            )
+            if self._optimizer_in_bwd:
```

Comment:
nit: should put `if not self._optimizer_in_bwd` first? the common case should be surfaced to readers first. but this is totally myt personal style